manila/share/drivers/scality/driver.py

```diff
@@ -191,26 +324,88 @@ def delete_share(self, context, share, share_server=None):
 
-        stats = {
-            'share_backend_name': backend_name,
-            'vendor_name': 'Scality',
-            'storage_protocol': self.NFS_PROTOCOL,
-            'driver_version': self.VERSION,
-        }
+        try:
+            self._management_call(command)
+        except processutils.ProcessExecutionError as e:
+            if e.exit_code == self.EXPORT_EXISTS:
+                msg = _("Share '%(name)s' (%(id)s) already defined.") % {
+                    'name': share['name'], 'id': share['id']}
+                raise exception.ShareBackendException(msg=msg)
+            else:
+                raise
 
-        super(ScalityShareDriver, self)._update_share_stats(stats)
+        return self._location_from_id(share['id'])
 
-    def _enforce_ip_acl(self, access):
-        """Check that the access is IP based."""
-        if access['access_type'] != 'ip':
-            raise exception.ManilaException('Only IP restrictions supported')
+    def ensure_share(self, share):
+        # Export locations are derived from the `export_ip` configuration
+        # parameter, and may thus change between service restarts. It is
+        # therefor always returned here if the share exists.
+        try:
+            self._management_call('get %s' % share['id'])
+
+        except processutils.ProcessExecutionError as e:
+            if e.exit_code == self.EXPORT_NOT_FOUND:
+                msg = _("'%(name)s' (%(id)s) not found") % {
+                    'name': share['name'], 'id': share['id']}
+                raise exception.InvalidShare(reason=msg)
+
+            else:
+                raise
+
+        return self._location_from_id(share['id'])
+
+    def _location_from_id(self, share_id):
+        """Format an export location from a share_id.
+
+        :param share_id: share id to format
+        :type share_id: string
+        :returns: string
+        """
+        raise NotImplementedError()
+
+
+class CIFSHelper(NASHelperBase):
+
+    PROTOCOL = "SMB"
+
+    def __init__(self, ssh_pool, export_ip, export_root):
+        super(CIFSHelper, self).__init__(ssh_pool, export_ip)
+        self.optional_args = "--root-export %s" % export_root
+
+    def allow_access(self, share, access):
+        if access['access_level'] != const.ACCESS_LEVEL_RW:
+            raise exception.InvalidShareAccessLevel(
+                level=access['access_level'])
+
+        access = copy.deepcopy(access)
+        # The scality-manila-utils implicitely set the access level to RW
+        # Hence, it doesn"t expect another CLI argument
+        access['access_level'] = ""
+        super(CIFSHelper, self).allow_access(share, access)
+
+    def deny_access(self, share, access):
+        if access['access_level'] != const.ACCESS_LEVEL_RW:
+            return
+
+        super(CIFSHelper, self).deny_access(share, access)
+
+    def _location_from_id(self, share_id):
+        return '\\\\%s\\%s' % (self.export_ip, share_id)
+
+
+class NFSHelper(NASHelperBase):
```

Comment:
What in the `NASHelperBase` class leads you think it's more a NFSHelper ? I don't see any reference to NFS in the `NASHelperBase`

If I define a "base class" with only abstract methods, I would have to duplicate a lot of code in the NFS and CIFS class because they really look the same.